rules-testing/test/recipes-test.js

```diff
@@ -0,0 +1,236 @@
+const assert = require('assert');
+const firebase = require("@firebase/testing");
+// var expect = require('chai').expect;
+
+const MY_PROJECT_ID = "boq-devex-step-2020";
+
+const currentUser = {
+  uid: "user_erica",
+};
+
+const otherUser = {
+  uid: "user_not_erica",
+};
+
+const myAuth = {uid: currentUser.uid};
+
+let validRecipe = {
+  recipeName: "Vanilla Cake",
+  uploaderUid: currentUser.uid,
+  difficulty: "Beginner",
+  description: "This is a cake. I really want to eat it.",
+  ingredients: ["cake", "frosting", "sprinkles"],
+  tools: ["fork", "knife"],
+  images: ["default image"],
+  instructions: ["eat cake"],
+  extraInfo: "",
+  timestamp: 2,
+  ratings: [],
+  tags: ["dessert"],
+};
+
+const invalidRecipe = {
+  recipeName: "Unicorn Cake",
+};
+
+const invalidDataRecipe = {
+  recipeName: 4,
+  uploaderUid: currentUser.uid,
+  difficulty: "Beginner",
+  description: "This is a cake. I really want to eat it.",
+  ingredients: ["cake", "frosting", "sprinkles"],
+  tools: ["fork", "knife"],
+  images: ["default image"],
+  instructions: ["eat cake"],
+  extraInfo: "",
+  timestamp: 2,
+  ratings: [],
+  tags: ["dessert"],  
+};
+
+const noIngredientsRecipe = {
```

Comment:
I'm not sure how to do a deep copy on another recipe instance, but if it was easy then this could be simplified to just make a deep copy of `validRecipe` and then clear out the ingredients. Just FYI